prover/maths/field/bls12_377.go

```diff
@@ -157,3 +157,67 @@ func FromBool(b bool) Element {
 	}
 	return Zero()
 }
+
+// ExpVec sets vector[i] = a[i]ᵏ for all i
+func ExpVec(vector, a Vector, k int64) {
+	N := len(a)
+	if N != len(vector) {
+		panic("vector.Exp: vectors don't have the same length")
+	}
+	if k == 0 {
+		for i := range vector {
+			vector[i].SetOne()
+		}
+		return
+	}
+	base := a
+	exp := k
+	if k < 0 {
+		// call batch inverse
+		base = BatchInvert(a)
+		exp = -k // if k == math.MinInt64, -k overflows, but uint64(-k) is correct
+	} else if N > 0 {
+		// ensure that vector and a are not the same slice; else we need to copy a into base
+		v0 := &vector[0] // #nosec G602 we check that N > 0 above
+		a0 := &a[0]      // #nosec G602 we check that N > 0 above
+		if v0 == a0 {
+			base = make(Vector, N)
+			copy(base, a)
+		}
+	}
+
+	copy(vector, base)
```

Comment:
I think this comes from bavard, but imo we could have a version when `a==vector` where we don't do two copies, but instead directly copy `copy(vector, a)`.